plugin-server/src/worker/ingestion/action-matcher.ts

```diff
@@ -160,20 +200,61 @@ export class ActionMatcher {
     }
 
     /**
-     * Base level of action matching.
+     * Base level of action matching (HogVM version).
      *
      * Return whether the event is a match for the action.
      * The event is considered a match if any of the action's steps (match groups) is a match.
      */
-    public async checkAction(
+    public checkActionBytecode(
+        event: PostIngestionEvent,
+        elements: Element[] | undefined,
+        action: Action
+    ): boolean | undefined {
+        try {
+            if (Array.isArray(action.bytecode) && action.bytecode.length > 1) {
+                const eventProxy = new Proxy(event, {
```

Comment:
I did this way because `elementsToString` feels like a complex operation, which we can opt out of performing on this hot path, unless something needs it.